### PR TITLE
Location & Event routes

### PR DESCRIPTION
**Pull request by tarungog**
Fixes # .
284 285
### PR Description
Creates event and location express routes
### What needs to be reviewed?
Is the Event interface what we want it to be? currently the response json is an unedited document straight from mongoose so it only contains the _id of a location rather than the actual location data. I'm not sure what the intension is.
### Context?
Lets map dev people start working

### CODE REVIEW

**Comment by tarungog** [comment]
@connorkrupp merge away

File: server/routes/api.js [hunk 1]
```diff
@@ -33,6 +35,7 @@ router.use('/reader', authMiddleware('admin reader', 'api'), readerHandler);
 router.use('/push', pushHandler);
 router.use('/scan', scanHandler);
 router.use('/sponsor', sponsorHandler);
+router.use('/location', locationHandler), router.use('/event', eventHandler);
```
Comment (antoniomika) [inline server/routes/api.js]: Let's separate these

Comment (antoniomika) [inline server/routes/api.js]: Let's separate these

File: server/routes/api/event.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                    .exec()
+                    .then(location => {
+                        req.body.location = location._id;
+                        return Event.create({
+                            name: req.body.name,
+                            longitude: req.body.longitude,
+                            latitude: req.body.latitude
+                        });
+                    })
+                    .then(event => {
```
Comment (antoniomika) [inline server/routes/api/event.js]: I like the promise chaining 👍

File: server/routes/api/event.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            res.status(500).send({
+                status: false,
+                message: Responses.UNKNOWN_ERROR
+            });
+        });
+});
+
+// Handles /v1/event/
+router.get('/', function(req, res) {
+    console.log(req.params);
```
Comment (antoniomika) [inline server/routes/api/event.js]: Is this needed?

File: server/routes/api.js (hunk 1 above)
Comment (tarungog) [reply server/routes/api.js]: @antoniomika i'm not sure why but prettify keeps turning this into one line -- not sure what to do about that

Comment (antoniomika) [reply server/routes/api.js]: Even where there is a `;` in between?

**Comment by tarungog** [comment]
@antoniomika fixed everything

**Comment by antoniomika** [comment]
Hmm looks like that merge conflict you resolved removed some stuff that's actually important. Want to take another go at that?

**Pull request merged**